### PR TITLE
feat: allow passing Floating UI's `FlipOptions` to `positioning.flip` property 

### PR DESCRIPTION
**Pull request by msigwart**
### Motivation 
Currently when using builders like `createDropdownMenu`, `positioning.flip` is a boolean property. Meaning flipping of the menu in case of overflow can either be turned on or off.

**With `flip: false`:**

https://github.com/melt-ui/melt-ui/assets/9678787/33425aae-505b-4204-9b0a-df110dfb7a30

**With `flip: true`:**

https://github.com/melt-ui/melt-ui/assets/9678787/692c567d-96d5-41cb-b04a-839d3ef8e109

Notice how with `flip: true`  the menu is flipped on both axes in case of overflow.

It would be great if it was possible to customize the flip behaviour further than just turning it on or off.

### Changes
This PR extends the flip options to optionally take an object with [flip customization options offered by Floating UI](https://floating-ui.com/docs/flip#options). This allows more fine-grained control over the flip behaviour of menus. 

Screenshot 2024-05-15 at 15 57 01@2x

For example, it now becomes possible to turn flipping _on_ for the main axis but _off_ for the cross axis. 

```svelte
const {
    elements: { subMenu, subTrigger },
    states: { subOpen },
} = createSubmenu({
    positioning: {
	placement: 'right-start',
        flip: {
	    mainAxis: true,
	    crossAxis: false,
        },
    },
});
```

https://github.com/melt-ui/melt-ui/assets/9678787/d7cedd11-1a4e-4cbf-a9a9-6008e9a7d8a7

Notice, how in the video the submenu is opened on the left instead of the right (because of the right-hand overflow on the "main" axis) but only shifted upwards instead of flipped upward (because of the bottom overflow on the "cross" axis).

### Testing
I've temporarily adjusted [the main preview of the dropdown menu component](http://localhost:5173/docs/builders/dropdown-menu#menu) to showcase the new behaviour. 

You can adjust the browser window so that the opening the dropdown menu would cause an overflow and see the new behaviour in action. 

---
We've recently started using Melt UI and really enjoy it so far so curious to hear your feedback on this PR!

### CODE REVIEW

**Comment by TGlide** [comment]
Hey, this is great stuff! Thank you!

File: src/lib/internal/actions/floating/action.ts [hunk 1]
```diff
@@ -38,6 +38,8 @@ const ALIGN_OPTIONS = ['start', 'center', 'end'] as const;
 type Side = (typeof SIDE_OPTIONS)[number];
 type Align = (typeof ALIGN_OPTIONS)[number];
 
+const isFlipOptions = (value: any): value is FlipOptions => typeof value === 'object';
```
Comment (TGlide) [inline src/lib/internal/actions/floating/action.ts:41]: This type-guard is misleading, and can lead to errors in the future. Wouldn't an `isObject` type guard suffice? IIRC we have one already.

File: src/docs/components/preview-wrapper.svelte [hunk 2]
```diff
@@ -34,7 +34,7 @@
 </script>
 
 <div class={cn(previewVariants({ variant, size }))} data-variant={variant} {id}>
-	<div class={cn('z-10 mx-auto inline-block px-4')}>
+	<div class={cn('z-10 mb-4 ml-auto inline-block self-end px-4')}>
```
Comment (TGlide) [inline src/docs/components/preview-wrapper.svelte:37]: Please change this before merging :)

**Comment by TGlide** [comment]
Also, it seems like I can't edit the PR directly. Could you add a changeset with `npx changeset`?

File: src/lib/internal/actions/floating/action.ts (hunk 1 above)
Comment (msigwart) [reply src/lib/internal/actions/floating/action.ts]: Good idea, replaced with `isObject`.

File: src/docs/components/preview-wrapper.svelte (hunk 2 above)
Comment (msigwart) [reply src/docs/components/preview-wrapper.svelte]: Done!

**Comment by msigwart** [comment]
Thanks for the review, I've added a changeset and I think otherwise the PR should be ready.

**Review by TGlide — APPROVED** [review]
Thank you!

**Pull request merged**